### ISSUE TITLE
Misleading error: could not determine a default font for sans-serif

### ISSUE THREAD

**Issue by zabackary** [slint-ui/slint#9879]
### Bug Description

When building a project with the `material-dark` style and the `Roboto` font not installed, the build fails with the following error:

```
   --> /home/username/projects/project/ui/MainWindow.slint:1:1
    |
  1 | import { AboutSlint, VerticalBox, HorizontalBox, Palette, Button, GroupBox, TabWidget } from "std-widgets.slint";
    | ^


  thread 'main' (239825) panicked at build.rs:8:6:
  Slint build failed: CompileError(["/home/username/projects/project/ui/MainWindow.slint:1: internal error: could not determine a default font for sans-serif"])
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

However, as suggested in #1240, I do have a `sans-serif` font installed. However, I do not have Roboto installed. Tracking down the source of the error, I can see that it occurs if `source_location` is not set. However, `source_location` is looked up from the `default-font-family` binding, which is set to `Roboto` (not `sans-serif`) in the case of material-dark.

This is a bit of a misleading error (it let me on a wild goose chase trying to fix my build environment and fonts) so it might be better to try to identify the culprit missing font.

See: https://github.com/slint-ui/slint/blob/e2670075fa19134aaf99bd75a848055209bed947/internal/compiler/passes/embed_glyphs.rs#L160

Related: #1240

### Reproducible Code (if applicable)

```slint

```

### Environment Details

- Slint Version: 1.14.1
- Platform/OS:
    - Target: Custom platform (`armv7a-vex-v5`) with `EmbedForSoftwareRenderer` set
    - Host (more applicable): NixOS
- Programming Language: Rust
- Backend/Renderer: Software

### Product Impact

Not a blocker because it is easy for me to install Roboto.

Thank you for developing Slint! The DX is amazing.

**Comment by tronical** [comment]
Reproducer:

```slint
export component App inherits Window {
    Text {
        text: "ok";
    }
}
```

Run with

```bash
cargo run -p slint-compiler -- --embed-resources=embed-for-software-renderer --style=material -f rust test case.slint
```
